x-pack/plugins/ml/public/application/components/navigation_menu/date_picker_wrapper/date_picker_wrapper.tsx

```diff
@@ -65,11 +70,33 @@ export const DatePickerWrapper: FC = () => {
 
   const { timefilter, history } = services.data.query.timefilter;
 
+  const { displayWarningToast } = useToastNotificationService();
+
   const [globalState, setGlobalState] = useUrlState('_g');
   const getRecentlyUsedRanges = getRecentlyUsedRangesFactory(history);
 
-  const refreshInterval: RefreshInterval =
-    globalState?.refreshInterval ?? timefilter.getRefreshInterval();
+  const timeFilterRefreshInterval = useRefreshIntervalUpdates();
+
+  const refreshInterval = useMemo((): RefreshInterval => {
+    const resultInterval = globalState?.refreshInterval ?? timeFilterRefreshInterval;
+
+    let value = resultInterval.value;
+    if (resultInterval.value < DEFAULT_REFRESH_INTERVAL_MS) {
+      displayWarningToast(
+        i18n.translate('xpack.ml.datePicker.shortRefreshIntervalWarningMessage', {
+          defaultMessage: 'Configured refresh interval is too short.',
```

Comment:
@lcawl do we need to mention that the page will use the default interval instead?

Reply:
What is the "default" interval (i.e. is it timepicker:refreshIntervalDefaults or will there be something like securitySolution:refreshIntervalDefaults that is specific to the ML app)? If it's possible to be explicit about what value we'll use, I think would be clearest. For example:

```suggestion
          defaultMessage: 'Configured refresh interval is too short, a value of 1s will be used instead.',
```

Reply:
@lcawl We resolve the refresh interval with the following order

1. Get it from the URL state 
2. If not provided in the URL state, get `timepicker:refreshIntervalDefaults` from Kibana settings 
3. Check if the resulting interval is too low and set at least 1s

Now I'm thinking if we should be explicit about which refresh interval is too short, i.e.

- Refresh interval provided with the URL state is too short 
- Configured refresh interval (`timepicker:refreshIntervalDefaults`) is too short 

🤔

Reply:
The text has been changed in https://github.com/elastic/kibana/pull/124851/commits/39f50d2123f329697bf333b168e96927b50c8783

Reply:
@lcawl I wonder if a bit more clarification would be helpful here, such as 
```
The refresh interval configured under Advanced Settings is shorter than the minimum supported by Machine Learning. A value of 10s will be used instead. 
```
or
```
The refresh interval configured in the URL is shorter than the minimum supported by Machine Learning. A value of 10s will be used instead. 
```

Reply:
Personally I think I'd find that level of detail useful, in case I want to go and change the way it's currently set.  If it's possible for the messages to be that specific, I like your suggestion, though I think "configured" is unnecessary. For example:

"The refresh interval in Advanced Settings is ..." and "The refresh interval in the URL..."

Reply:
Text updated in https://github.com/elastic/kibana/pull/124851/commits/cbddd67f16bce30b5090343442d51b004b86bfc9